webrender_api/src/display_list.rs

```diff
@@ -771,6 +771,9 @@ impl DisplayListBuilder {
     }
 
     fn push_item(&mut self, item: SpecificDisplayItem, info: &LayoutPrimitiveInfo) {
+        if info.rect.is_empty() {
```

Comment:
There's a couple of non-visible items that use push_item - e.g. sticky frame and push_shadow. Are those both fine to skip if the rect is empty?

Reply:
I think all the stacking contexts that gecko pushes also have empty rects, so this is probably not safe

Reply:
stacking context are not affected by this commit but if sticky frames and shadows need to work with empty rects then my naïve stab at this won't do. I can also filter empty rects out on a per item basis (in which case doing this on the wr or gecko side is the same amount of effort so don't hesitate to let me know if you feel like it should be the caller's responsibility.

Reply:
I *think* that it might be an issue for shadows - like pictures, we calculate the required rect based on the contents placed in the shadow. I'm not sure about sticky frames - @mrobinson may know. I do think it's a good idea to handle this in WR, since it can actually cause visual bugs if we try to process an item with size < 0.